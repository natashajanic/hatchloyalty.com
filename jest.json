{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
