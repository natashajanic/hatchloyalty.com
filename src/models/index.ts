import { FluidObject } from 'gatsby-image'

export interface IBlogPost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    author: string
    date: string
    featuredImage?: IGraphQLImage
    tags: string[]
    title: string
  }
  html: string
}

export interface IGraphQLImage {
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IRelease {
  changes: Array<{ component: string; description: string; link?: string }>
  date: string
  name: string
}

export interface IResourcePost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    title: string
  }
  html: string
}

export interface ITeamMember {
  headshot?: IGraphQLImage
  name: string
  position: string
}
