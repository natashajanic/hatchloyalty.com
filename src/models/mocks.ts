import { IRelease, IBlogPost, IResourcePost, ITeamMember } from '.'

export const mockBlogPost: IBlogPost = {
  excerpt: 'This is the excerpt for a blog post.',
  fields: {
    slug: '/2019-02-25-test-blog-post/',
  },
  frontmatter: {
    author: 'John Doe',
    date: 'February 25, 2019',
    featuredImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1,
          src: 'test/imgs/test-blog-post.png',
          srcSet: '',
          sizes: '',
        },
      },
    },
    tags: ['Test', 'Press Release'],
    title: 'Test Blog Post',
  },
  html: '<div>Test blog post body content.</div>',
}

export const mockRelease: IRelease = {
  changes: [
    { component: 'Test', description: 'Test Change 1' },
    {
      component: 'Test HQ',
      description: 'Test Change 2',
      link: 'https://developer.hatchloyalty.com/test/change-2',
    },
  ],
  date: 'January 2019',
  name: 'Test Release 2019.01',
}

export const mockResourcePost: IResourcePost = {
  excerpt: 'This is the excerpt for a resource post.',
  fields: {
    slug: '/test-resource-post/',
  },
  frontmatter: {
    title: 'Test Resource Post',
  },
  html: '<div>Test resource post body content.</div>',
}

export const mockTeamMember: ITeamMember = {
  headshot: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: 'test/imgs/john-doe.png',
        srcSet: '',
        sizes: '',
      },
    },
  },
  name: 'John Doe',
  position: 'CEO',
}
