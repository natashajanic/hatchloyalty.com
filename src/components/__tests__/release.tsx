import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Badge from 'src/components/badge'
import { mockRelease } from 'src/models/mocks'
import Release, { IReleaseProps } from '../release'

describe('Release', () => {
  let props: IReleaseProps
  let mountedComponent: ShallowWrapper<any, any, Release>

  beforeEach(() => {
    props = {
      release: mockRelease,
    }
    mountedComponent = shallow(<Release {...props} />)
  })

  describe('render', () => {
    it('renders the name of the release', async () => {
      expect(mountedComponent.find('Text[as="h3"]').length).toEqual(1)
      expect(mountedComponent.find('Text[as="h3"]').text()).toEqual(
        props.release.name
      )
    })

    it('renders each change contained within the release', async () => {
      expect(mountedComponent.find(Badge).length).toEqual(
        props.release.changes.length
      )
      props.release.changes.forEach((change, index) => {
        expect(
          mountedComponent
            .find(Badge)
            .at(index)
            .text()
        ).toEqual(change.component)
        expect(mountedComponent.text()).toContain(change.description)
      })
    })

    it('renders a link for any change that has one specified', async () => {
      mountedComponent.setProps({
        release: {
          ...mockRelease,
          changes: [
            {
              component: 'Test',
              description: 'Test Change',
              link: 'https://developer.hatchloyalty.com/test-change',
            },
          ],
        },
      })
      expect(mountedComponent.find('a').length).toEqual(1)
      const changeLink = mountedComponent.find('a')
      expect(changeLink.prop('href')).toEqual(
        'https://developer.hatchloyalty.com/test-change'
      )
      expect(changeLink.text()).toEqual('Click to read more')
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
