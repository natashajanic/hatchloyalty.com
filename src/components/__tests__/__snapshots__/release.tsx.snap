// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release render matches the expected snapshot 1`] = `
<Panel
  blacklist={
    Array [
      "cursor",
    ]
  }
  border="1px solid"
  borderColor="grayLight"
  borderRadius={4}
  m={0}
  mb={5}
  mt={3}
  p={0}
  width="100%"
>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "border",
        "borderTop",
        "borderRight",
        "borderBottom",
        "borderLeft",
        "borderColor",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    borderBottom="1px solid"
    borderColor="grayLight"
    p={3}
  >
    <Text
      as="h3"
      blacklist={
        Array [
          "css",
          "fontSize",
          "lineHeight",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "textStyle",
          "alignItems",
          "color",
          "bg",
          "display",
          "flex",
          "fontWeight",
          "textAlign",
        ]
      }
      fontSize={3}
      lineHeight="1.5"
      m={0}
      my={1}
      textAlign="center"
      textStyle="normal"
    >
      <ForwardRef
        to="/releases/test-release-2019-01"
      >
        Test Release 2019.01
      </ForwardRef>
    </Text>
  </Styled(Component)>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "border",
        "borderTop",
        "borderRight",
        "borderBottom",
        "borderLeft",
        "borderColor",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    borderBottom="1px solid"
    borderColor="grayLight"
    key="change-0"
    p={3}
  >
    <Styled(Component)
      alignItems="flex-start"
      blacklist={
        Array [
          "css",
          "display",
          "flexWrap",
          "alignItems",
          "flexDirection",
          "justifyContent",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "width",
        ]
      }
      display="flex"
      flexWrap="wrap"
    >
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "display",
            "flexWrap",
            "alignItems",
            "flexDirection",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        display="flex"
        flexWrap="wrap"
        justifyContent="flex-end"
        width={120}
      >
        <Styled(Component)
          alignItems="center"
          background="greenXLight"
          blacklist={
            Array [
              "css",
              "alignItems",
              "background",
              "border",
              "borderTop",
              "borderRight",
              "borderBottom",
              "borderLeft",
              "borderColor",
              "borderRadius",
              "color",
              "bg",
              "display",
              "fontSize",
              "fontWeight",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
          border="1px solid"
          borderColor="green"
          borderRadius="10%"
          color="green"
          display="flex"
          fontSize={1}
          fontWeight="bold"
          justifyContent="center"
          mx={2}
          px={2}
        >
          Test
        </Styled(Component)>
      </Styled(Component)>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        ml={2}
        style={
          Object {
            "flex": "1 1 0%",
          }
        }
      >
        <Text
          blacklist={
            Array [
              "css",
              "fontSize",
              "lineHeight",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "textStyle",
              "alignItems",
              "color",
              "bg",
              "display",
              "flex",
              "fontWeight",
              "textAlign",
            ]
          }
          fontSize={3}
          lineHeight="1.5"
          m={0}
          textStyle="normal"
        >
          Test Change 1
        </Text>
      </Styled(Component)>
    </Styled(Component)>
  </Styled(Component)>
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "border",
        "borderTop",
        "borderRight",
        "borderBottom",
        "borderLeft",
        "borderColor",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    borderBottom="1px solid"
    borderColor="grayLight"
    key="change-1"
    p={3}
  >
    <Styled(Component)
      alignItems="flex-start"
      blacklist={
        Array [
          "css",
          "display",
          "flexWrap",
          "alignItems",
          "flexDirection",
          "justifyContent",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "width",
        ]
      }
      display="flex"
      flexWrap="wrap"
    >
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "display",
            "flexWrap",
            "alignItems",
            "flexDirection",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        display="flex"
        flexWrap="wrap"
        justifyContent="flex-end"
        width={120}
      >
        <Styled(Component)
          alignItems="center"
          background="greenXLight"
          blacklist={
            Array [
              "css",
              "alignItems",
              "background",
              "border",
              "borderTop",
              "borderRight",
              "borderBottom",
              "borderLeft",
              "borderColor",
              "borderRadius",
              "color",
              "bg",
              "display",
              "fontSize",
              "fontWeight",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
          border="1px solid"
          borderColor="green"
          borderRadius="10%"
          color="green"
          display="flex"
          fontSize={1}
          fontWeight="bold"
          justifyContent="center"
          mx={2}
          px={2}
        >
          Test HQ
        </Styled(Component)>
      </Styled(Component)>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        ml={2}
        style={
          Object {
            "flex": "1 1 0%",
          }
        }
      >
        <Text
          blacklist={
            Array [
              "css",
              "fontSize",
              "lineHeight",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "textStyle",
              "alignItems",
              "color",
              "bg",
              "display",
              "flex",
              "fontWeight",
              "textAlign",
            ]
          }
          fontSize={3}
          lineHeight="1.5"
          m={0}
          textStyle="normal"
        >
          Test Change 2
        </Text>
        <ForwardRef
          to="https://developer.hatchloyalty.com/test/change-2"
        >
          <Text
            blacklist={
              Array [
                "css",
                "fontSize",
                "lineHeight",
                "m",
                "mt",
                "mr",
                "mb",
                "ml",
                "mx",
                "my",
                "p",
                "pt",
                "pr",
                "pb",
                "pl",
                "px",
                "py",
                "textStyle",
                "alignItems",
                "color",
                "bg",
                "display",
                "flex",
                "fontWeight",
                "textAlign",
              ]
            }
            fontSize={3}
            lineHeight="1.5"
            m={0}
            textStyle="normal"
          >
            Click to read more
          </Text>
        </ForwardRef>
      </Styled(Component)>
    </Styled(Component)>
  </Styled(Component)>
</Panel>
`;
